apps/betterangels-backend/accounts/models.py

```diff
@@ -79,11 +80,14 @@ class User(AbstractBaseUser, PermissionsMixin):
     servicerequestuserobjectpermission_set: models.QuerySet["ServiceRequestUserObjectPermission"]
 
     def __str__(self: "User") -> str:
-        return self.email
+        return f"{self.full_name} ({self.pk})"
 
     @model_property
     def full_name(self: "User") -> str:
-        return f"{self.first_name} {self.last_name}"
+        name_parts = filter(None, [self.first_name, self.middle_name, self.last_name])
+        display_name = " ".join(name_parts).strip()
```

Comment:
Code Golf not a requirement for change, but useful to know since I've seen us use this pattern in other places.:
This can be simpler if you want to use [compact from funcy](https://funcy.readthedocs.io/en/stable/colls.html?highlight=compact#compact)

```python
from funcy import compact
display_name = " ".join(compact([self.first_name, self.middle_name, self.last_name])).strip()
```

```python
from itertools import filterfalse
display_name = " ".join(filterfalse(bool, [self.first_name, self.middle_name, self.last_name])).strip()
```

Reply:
funcy is not a dependency, and using itertools.filterfalse causes an `arg-type` typecheck error:
```
 error: Argument 2 to "filterfalse" has incompatible type "List[Optional[str]]"; expected "Iterable[str]"  [arg-type]
```